Gutenberg: Calypsoify links to the editor

Copons opened the pull request:
Follow up to #28216
Alternative to #28277 

#### Changes proposed in this Pull Request

* Calypsoify some links to the editor in order to save one unnecessary redirection.
* Disable the "Duplicate" feature for all post types because in Calypso works the opposite way than WP-Admin: Calypso copies the selected post into a new post; WP-Admin copies a selected post into the current post.

#### Testing instructions

- Opt in to Gutenberg by entering this into the browser console:
```es6
dispatch( { type: 'EDITOR_TYPE_SET', siteId: state.ui.selectedSiteId, editor: 'gutenberg' } )
```

- Make sure that the "Add" buttons in the Sidebar opens the Calypsoified Gutenberg for posts, pages, and CPTs.

- Open `/posts`, `/pages` and a CPT list such as `/types/feedback`, and make sure that:
  -  The post (or page, or CPT) item's title opens the Calypsoified Gutenberg (and its `href` contains `wp-admin`).
  - The "Edit" button in the post item's actions popover (aka the ellipsis menu) opens the Calypsoified Gutenberg.
  - The post item's action popover doesn't contain any "Duplicate" or "Copy" items.

vindl commented on client/my-sites/sidebar/manage-menu.jsx:
```diff
@@ -71,8 +73,22 @@ class ManageMenu extends PureComponent {
 		return allSingleSites ? '' : '/my';
 	}
 
+	getEditorUrl = postType => {
```
Why are we redeclaring this function here instead of reusing `getEditorUrl` from `state/selectors`?

Copons replied:
Because it needs to run for each post type, and I think it's more appropriate to do so when needed rather than in `mapStateToProps`.
But then again, I suppose I could also do something like this as well:
```es6
// mapStateToProps
editorUrls: reduce( postTypes, ( editorUrls, postType ) => {
  editorUrls[ postType ] = editorUrl( state, siteId, null, postType );
  return editorUrls;
}, {} ),
```

vindl commented:
I think I found an edge case with CPTs when the list of posts is empty:

<img width="444" alt="screenshot 2018-11-05 at 22 56 27" src="https://user-images.githubusercontent.com/1182160/48029910-c0e99680-e14f-11e8-9f39-0cfb87ed19f2.png">

`Add New Testimonial` is still linking to Calypso. This is not a blocker though, feel free to address it in a follow-up if you prefer so.

gwwar commented on client/state/selectors/get-editor-url.js:
```diff
@@ -0,0 +1,38 @@
+import { getEditorPath } from 'state/ui/editor/selectors';
+
+export const getEditorUrl = ( state, siteId, postId = null, postType = 'post' ) => {
+	const calypsoifyGutenberg = isEnabled( 'calypsoify/gutenberg' );
```
We probably also want to turn this off for Jetpack/VIP until we're ready to include them. 

I did something similar here for opt-in flows:
https://github.com/Automattic/wp-calypso/pull/28252/files#diff-c0750dfc4c46c43a7d5d6ffa8d810463R70

```jsx
	const siteId = getSelectedSiteId( state );
	const isVip = isVipSite( state, siteId );
	const isJetpack = isJetpackSite( state, siteId );
    const isCalypsoifyGutenbergEnabled =  isEnabled( 'calypsoify/gutenberg' ) && ! isJetpack && ! isVip;
```

gwwar replied:
Fine to address this in a follow up PR, but let's not forget before release.

Copons replied:
Oh I believe this is one of those better safe than sorry cases. :)

Copons commented:
@gwwar @vindl I'd love one last look at this! 🙇 

In bd1750c I've moved the Calypsoify Gutenberg check into its own selector and added the VIP and JP checks as well.
In 612c5d0 I've "calypsoified" the "New" button of posts, pages, and CPTs `EmptyContent`.
